lua/terrortown/menus/gamemode/equipment/base_equipment.lua

```diff
@@ -5,100 +5,110 @@ local TryT = LANG.TryTranslation
 CLGAMEMODESUBMENU.priority = 0
 CLGAMEMODESUBMENU.title = ""
 
-local function Invert(data, value)
-	if data.inverted then
-		return not value
-	else
-		return value
-	end
-end
-
 function CLGAMEMODESUBMENU:Populate(parent)
 	local equipment = self.equipment
-	local forms = {}
-	local form
-	local masterRefs = {}
+	local accessName = ShopEditor.accessName
+	local itemName = equipment.id
+
+	if not self.isItem then
+		local form = vgui.CreateTTT2Form(parent, "header_equipment_weapon_spawn_setup")
 
-	for key, data in SortedPairsByMemberValue(ShopEditor.savingKeys, "order", false) do
-		if self.isItem and not data.showForItem then continue end
+		form:MakeHelp({
+			label = "equipmenteditor_desc_auto_spawnable"
+		})
 
-		form = forms[data.group or 1]
+		local master = form:MakeCheckBox({
+			label = "equipmenteditor_name_auto_spawnable",
+			database = {name = accessName, itemName = itemName, key = "AutoSpawnable"}
+		})
 
-		if not form then
-			form = vgui.CreateTTT2Form(parent, ShopEditor.groupTitles[data.group])
-			forms[data.group or 1] = form
+		local entType
+		local entTypeList = entspawnscript.GetEntTypeList(SPAWN_TYPE_WEAPON, {[WEAPON_TYPE_RANDOM] = true})
+		local choices = {}
+		for i = 1, #entTypeList do
+			entType = entTypeList[i]
+			choices[i] = {title = TryT(entspawnscript.GetLangIdentifierFromSpawnType(SPAWN_TYPE_WEAPON, entType)), value = entType}
 		end
 
-		local name = "equipmenteditor_name_" .. data.name
-		local desc = "equipmenteditor_desc_" .. data.name
+		form:MakeComboBox({
+			label = "equipmenteditor_name_spawn_type",
+			database = {name = accessName, itemName = itemName, key = "spawnType"},
```

Comment:
just an idea for better code docs: How about creating a function like the `Vector()` function that returns the desired object and also has default data. Something like this:

```lua
database = DatabaseElement(accessName, itemName, "spawnType")
```

Reply:
That sounds like a good idea, we could do this for serverConvar and convar as well. In general that could make sense to somewhere document the functionality of our UI tables, that we give to Make functions.

Reply:
I'm not that sure where to use this for normal convars as they only have a single value? but sure, do it!